react-native/react/more/dev/tracker-popup-listener.desktop.js

```diff
@@ -0,0 +1,309 @@
+/* @flow */
+
+// $FlowIssue with platform components
+import React, {Component} from '../../base-react'
+import Tracker from '../../tracker'
+
+import engine from '../../engine'
+import { createServer } from '../../engine/server'
+import { flattenCallMap, promisifyResponses } from '../../engine/call-map-middleware'
+
+import * as _ from 'lodash'
+
+import type { IdentifyKey, TrackSummary, Identity, RemoteProof, LinkCheckResult, Cryptocurrency, IdentifyOutcome, User, UserSummary} from '../../constants/types/flow-types'
+
+import { identify } from '../../keybase_v1.js'
+
+import type { BioProps } from '../../tracker/bio.render.desktop'
+import type { ActionProps } from '../../tracker/action.render.desktop'
+import type { HeaderProps } from '../../tracker/header.render.desktop'
+import type { ProofsProps, ProofsAndChecks } from '../../tracker/proofs.render.desktop'
+import type { TrackerProps } from '../../tracker/render.desktop'
+import type { SimpleProofState } from '../../tracker/common-types'
+
+type ComponentState = {
+  outstandingReqs: Array<RemoteProof>,
+  overallProofStatus: SimpleProofState,
+  bioProps: BioProps,
+  actionProps: ActionProps,
+  headerProps: HeaderProps,
+  proofsProps: ProofsProps
+}
+
+export default class TrackerPopupListener extends Component {
+
+  // TODO's
+  // * Get reason from the service
+  // * Get UserInfo (for BioProps)
+  //
+
+  state: ComponentState;
+
+  // Get some type safety here
+  setState (object: ComponentState): void {
+    super.setState(object)
+  }
+
+  constructor (props: {}) {
+    super(props)
+    const overallProofStatus = 'pending'
+    this.state = {
+      outstandingReqs: [],
+      overallProofStatus,
+      bioProps: {
+        username: null,
+        state: overallProofStatus,
+        userInfo: {
+          fullname: 'TODO: get this information',
+          followersCount: -1,
+          followingCount: -1,
+          followsYou: false,
+          avatar: 'TODO: get this information',
+          location: 'TODO: get this information'
+        }
+      },
+
+      actionProps: {
+        state: overallProofStatus,
+        username: null,
+        shouldFollow: true,
+        onClose: () => {
+          console.log('onClose')
+        }, // TODO
+        onRefollow: () => {
+          console.log('onRefollow')
+        },
+        onUnfollow: () => {
+          console.log('onUnfollow')
+        },
+        onFollowHelp: () => window.open('https://keybase.io/docs/tracking'), // TODO
+          // followChecked: checked => this.setState({shouldFollowChecked: checked})
+        followChecked: checked => console.log('follow checked:', checked)
+      },
+
+      headerProps: {
+        reason: 'TODO: get this information',
+        onClose: () => {
+          // TODO
+          console.log('onClose from header')
+        } 
+      },
+
+      proofsProps: {
+        proofsAndChecks: []
+      }
+    }
+  }
+
+  updateProofsAndChecks (rp: RemoteProof, lcr: LinkCheckResult): void {
```

Comment:
it seems like this and the 'todo move' stuff below really all belong in the action layer